docs/tutorials/getting_started.md

```diff
@@ -117,7 +117,7 @@ async def run() -> None:
     )
 
     # Define your application logic here
-    grid_meter = microgrid.logical_meter().grid_power.new_receiver()
+    grid_meter = microgrid.grid().power.new_receiver()
```

Comment:
@frequenz-floss/python-sdk-team the example here does not consider that `microgrid.grid()` can return `None`, should I amend the example? Or is it up to the user to check the documentation and add checks or error-handling as this is just an example?

Reply:
oh really?  I thought the stream at least should always be accessible.  Has nothing to do with whether metadata is available or not.  No grid metadata doesn't mean there's no grid component in the component graph, right?

And even if there's no grid component, the formula should still be available, so that people are able to build one general purpose algorithm, rather than one algorithm when grid power is available, one when it isn't, etc.

Reply:
In other words, I don't think we should put the grid formula into a class that can be None.  If we are doing this, then I think `grid` has to always be there.

Reply:
Then I'll need to work out the Grid interface, I think I can make the `Fuse` attribute optional instead of the `Grid` object so that the microgrids configured as islands won't have a grid connection (fuse == None). Another alternative would be to use another class

Reply:
Yeah, good point. As we mentioned earlier, if the idea is to abstract away when we are on an island, we might want to still have a (virtual?) fuse with a current rate of `0`. So you can treat the grid connection generically.

Reply:
But yeah, I can see how makes us re-think the grid connection interface, but I think it is going in a good direction.

Reply:
> we might want to still have a (virtual?) fuse with a current rate of 0

true, maybe worth making an issue, or picking up immediately after this.

Reply:
Added the commit _Refactor Grid interface_ but I can move it to a new PR if you think is better

Reply:
Sorry, I'm realizing it might be more complicated than that.  Or maybe I need to look into the current implementation more.  We always want to have a grid interface.  But not sure we'll always want to have a non-None fuse interface.  This is because the api might not give us fuse info,  but there would still be a grid component in the component graph.  

In that case,  using 0 value for fuse would be inaccurate, we just don't know the fuse value.  So we have to say it is None.

So there can be situations where fuse has to be None,  but will confirm on Monday.

Reply:
Even otherwise, the way to confirm that it is an island config, is by looking at the component graph, and not from the metadata.

Reply:
Is why dealing with it separately might be easier.  Although I might be completely wrong, haven't looked at the code yet.

Reply:
So no grid connection -> `fuse = 0`
Unknown fuse -> `fuse = None`, which in practice means infinite? Or what else can be done if we know there is a grid connection but have no fuse info or there is no fuse?

Reply:
Still, thinking about failure scenarios, if for some reason we don't receive a fuse (a bug in the microgrid service), we should still be able to operate, right? Only a few pieces might depend on the fuse info. So I think we should support that scenario, even if this force users to check if `fuse is not None` when then need to access the fuse info.

Reply:
> we should still be able to operate, right?

Yes, a lot of our actors don't depend on the fuse value, and I was not suggesting that we stop operating at all, just that we set `None` values for fuse, rather than 0.

And as you said, if an actor wants to explicitly protect against blowing the fuse, it can't.  In that scenario, it is up to the users to decide if they want to continue anyway or not.

Reply:
> Yes, a lot of our actors don't depend on the fuse value, and I was not suggesting that we stop operating at all, just that we set `None` values for fuse, rather than 0.

Oh, OK, then I misunderstood your "No", I guess it was only for the how to interpret the `None`, not the `None` itself :+1:

Reply:
The current source code on `v1.x.x` states that if there is grid connection then metadata and fuse must be available, otherwise it is a runtime error and the grid won't be initialized https://github.com/frequenz-floss/frequenz-sdk-python/blob/v1.x.x/src/frequenz/sdk/timeseries/grid.py#L60-L66

@tiyash-basu-frequenz can you please confirm that the metadata and fuse will be always set where there is a grid connection point? And is it a possibility that the fuse max current is unknown?

Reply:
Again, I wouldn't make the SDK depend on what the API currently does. Despite what we currently do, I think we should support cases where there is a grid connection but no fuse data for some reason. Just refuse to initialize the SDK because we don't have fuse info seems overkill.

Reply:
I agree with Luca. The SDK should just look at the API, and should not have any dependency on the service implementation.

Regarding fuses, it's actually very difficult to say. If a microgrid owner configures one, the API will expose it. Otherwise not.

Reply:
>Alternatively we can go to the edge team and ask them to confirm if we can always have this, because I don't see why not. Fuse limit for a location can't be a hard thing to find out.

We are not in control of this information. If a user provides this, we get it, otherwise we do not. If this blocks apps from running, then the app should issue an error, which will (in the future) get shown to the user in the UI.

Reply:
Thanks! Then I'll remove the original/initial fuse checks allowing it to be `None` and clarify that fuse could be `None` for either islanded microgrids or it is just unknown for any reason